File: 1108_defangIPaddr.py
class Solution:
    def defangIPaddr(self, address: str) -> str:
        defanged = ''
        for ch in address:
            if ch != '.':
                defanged += ch
            else:
                defanged += '[.]'
        return defanged

# Alternative solution

class Solution:
    def defangIPaddr(self, address):
        defanged = ""
        for char in address:
            defanged += char if char.isdigit() else '[.]'
        return defanged

File: 1001-1500/1108_defangIPaddr.java
class Solution {
    public String defangIPaddr(String address) {
        StringBuilder result = new StringBuilder();

        for (int i=0; i<address.length(); i++) {
            char c = address.charAt(i);
            if (c != '.') result.append(c);
            else result.append("[.]");
        }

        return result.toString();
    }
}
